istio-telemetry/kiali/values.yaml

```diff
@@ -1,15 +1,14 @@
-#
-# addon kiali
-#
 kiali:
-  enabled: false # Note that if using the demo or demo-auth yaml when installing via Helm, this default will be `true`.
-  replicaCount: 1
-  hub: quay.io/kiali
-  tag: v1.9
-  image: kiali
-  contextPath: /kiali # The root context path to access the Kiali UI.
+
+  enabled: false # If using demo profile, this will be set to `true`.
+
   nodeSelector: {}
   podAnnotations: {}
+  resources: {}
+  tolerations: []
+  replicaCount: 1
+
+  createDemoSecret: false # When true, a secret will be created with a default username and password. Useful for demos.
```

Comment:
We eventually want to get Istio to the state where everything is `true` - or turned on by default.  As this extends the API, can you give some thought to a better name for line 11 that matches with this goal?

Reply:
This was a setting introduced a while ago - @linsun asked for this because at the time when users were installing the demo profile, we were still requiring them to create a secret and some people forgot or didn't know and were not able to access Kiali. So it was decided to create a demo secret for the demo profile to avoid that problem. See this issue for background and discussion about this: https://github.com/istio/istio/issues/11244

So I don't know what we want to do here. We could remove it, and always require the user to create secrets - you can see where we tell people to do this in the Task doc here: https://istio.io/docs/tasks/observability/kiali/#create-a-secret

Or we could have a default authentication strategy that doesn't require a secret to contain credentials. In fact, Kiali does not require a secret by default if installing in OpenShift - Kiali just sets the auth.strategy to "openshift" which is an oauth mechanism to simply allow the user to log in using his/her OpenShift login credentials. There is another auth.strategy called "anonymous" which doesn't require the user to use any credentials - if a user has the Kiali URL, the user will gain access to the Kiali UI (so that also does not require a secret and may be a good alternative for the demo profile).